### PR TITLE
Add Dateutil to Dependencies and Add Date String Normalization 

### PR DESCRIPTION
**Pull request by IsaacMilarky**
## Problem

There is currently only simple normalization of date strings in dedupliFHIR.

## Solution

Add a date_normalization function to automatically normalize date strings to the same format (Y/m/d) 

## Result

The function `normalize_date_text` uses the `dateutil` library in order to automatically parse and normalize the dates that appear in the files that we are parsing. 

Remove unneeded dependencies as well. Also removed unneeded diff logic. 

## Test Plan
`make test`

### CODE REVIEW

File: cli/deduplifhirLib/normalization.py [hunk 1]
```diff
@@ -247,6 +248,19 @@ def british_to_american(input_text):
...
+    Normalizes the given date string
+
+    Arguments:
+        input_text: the input date text to normalize
+    
+    Returns:
+        The normalized date string
+    """
+    d = date_parser.parse(input_text)
+    return d.strftime("%Y-%m-%d")
```
Comment (patsier-cms) [inline cli/deduplifhirLib/normalization.py:262]: Should we handle exceptions here if it fails to parse? Not sure we'd run into it later, but what currently happens if someone has "TEST" in a birth date column?

**Review by patsier-cms — COMMENTED** [review]
One quick comment here, thanks!

File: cli/deduplifhirLib/normalization.py (hunk 1 above)
Comment (IsaacMilarky) [reply cli/deduplifhirLib/normalization.py]: I added an exception handle there to handle invalid date formats.

File: cli/deduplifhirLib/normalization.py [hunk 2]
```diff
@@ -247,6 +249,29 @@ def british_to_american(input_text):
...
+        d = date_parser.parse(input_text)
+    except ParserError as e:
+        print(f"Error when trying to parse date {input_text}")
+        print(f"Error: {e}")
+        return ""
+    except OverflowError as e:
+        print("Overflow error when trying to parse date!")
+        print("Input string too long!")
+        print(f"Error: {e}")
+        return ""
```
Comment (patsier-cms) [inline cli/deduplifhirLib/normalization.py:272]: Maybe we can just return the original string on the off-chance that it's close enough that Splink can match on that?

**Review by patsier-cms — COMMENTED** [review]
Last small item then I think this is good to go!

File: cli/deduplifhirLib/normalization.py (hunk 2 above)
Comment (IsaacMilarky) [reply cli/deduplifhirLib/normalization.py]: That makes sense.

**Review by patsier-cms — APPROVED** [review]
Looks great! Thanks for making those changes

**Pull request merged**